sacc2/sacc.py

```diff
@@ -0,0 +1,940 @@
+
+        **tags:
+            Tags to apply to this data point.
+            Tags can be any arbitrary metadata that you might want later,
+            For 2pt data the tag would include an angle theta or ell.
+
+        Returns
+        -------
+        None
+        """
+        if self.covariance is not None:
+            raise ValueError("You cannot add a data point after setting the covariance")
+        tracers = tuple(tracers)
+        for tracer in tracers:
+            if (tracer not in self.tracers) and (not tracers_later):
+                raise ValueError(f"Tracer named '{tracer}' is not in the known list of tracers."
+                                 "Either put it in before adding data points or set tracers_later=True")
+        d = DataPoint(data_type, tracers, value, **tags)
+        self.data.append(d)
+
+    def add_covariance(self, covariance):
+        """
+        Once you have finished adding data points, add a covariance
+        for the entire set.
+
+        Parameters
+        ----------
+        covariance: array or list
+            2x2 numpy array containing the covariance of the added data points
+            OR a list of blocks
+
+        Returns
+        -------
+        None
+        """
+        if isinstance(covariance, BaseCovariance):
+            self.covariance = covariance
+        else:
+            self.covariance = BaseCovariance.make(covariance, len(self))
+
+    def _indices_to_bool(self, mask):
+        # Convert an array of indices into a boolean True mask
+        if mask.dtype not in [np.int8, np.int16, np.int32, np.int64]:
+            raise ValueError(f"Wrong mask type ({mask.  dtype}) - expected integers or boolean")
+        m = np.zeros(len(self), dtype=bool)
+        for i in mask:
+            m[i] = True
+        return m
+
+    def mask_indices(self, indices):
+        """
+        Select data points, keeping only values where the mask is True or an index is
+        included in it.
+
+        You can use Sacc.cut_indices to do the opposite operation, keeping points where the mask
+        is False.
+
+        You use the Sacc.mask method to find indices and apply this method automatically,
+        or the Sacc.indices method to manually select indices.
+
+        Parameters
+        ----------
+        indices: array or list
+            Mask must be either a boolean array or a list/array of integer indices to remove.
+            if boolean then True means to keep a data point and False means to cut it
+            if integers then values indicate data points to keep
+        """
+        indices = np.array(indices)
+
+        # Convert integer masks to booleans
+        if indices.dtype != np.bool:
+            indices = self._indices_to_bool(indices)
+
+        self.data = [d for i, d in enumerate(self.data) if indices[i]]
+        if self.covariance is not None:
+            self.covariance = self.covariance.masked(indices)
+
+    def cut_indices(self, indices):
```

Comment:
This seems confusing to me. What is wrong with asking people to write `~mask` in the method before this one?